cunumeric/eager.py

```diff
@@ -460,20 +476,21 @@ def convert(self, rhs, warn=True):
                 else:
                     self.array[:] = rhs.array
 
-    def fill(self, value) -> None:
+    def fill(self, value: Any) -> None:
         if self.deferred is not None:
             self.deferred.fill(value)
         else:
             self.array.fill(value)
 
-    def dot(self, rhs1, rhs2):
+    def dot(self, rhs1: Any, rhs2: Any) -> None:
         self.check_eager_args(rhs1, rhs2)
         if self.deferred is not None:
-            self.deferred.dot(rhs1, rhs2)
+            # XXXX bug? deferred.dot does not exist
+            self.deferred.dot(rhs1, rhs2)  # type: ignore
```

Comment:
I think this is a bug? A the comment states, there is not `DeferredArray.dot`

Reply:
AFACIT the whole `EagerArray.dot` method is not used anymore, and can be removed.

Reply:
https://github.com/nv-legate/cunumeric/pull/438/commits/0d92336df3cd075ca63d64d90253ec93b08785d0